src/cython/cython/persistence_graphical_tools.py

```diff
@@ -318,8 +315,8 @@ def plot_persistence_density(persistence=[], persistence_file='',
                                  reverse=True)[:max_intervals]
 
         # Set as numpy array birth and death (remove undefined values - inf and NaN)
-        birth = np.asarray([(interval[1][0]) for interval in persistence_dim if (math.isfinite(interval[1][1]) and math.isfinite(interval[1][0]))])
-        death = np.asarray([(interval[1][1]) for interval in persistence_dim if (math.isfinite(interval[1][1]) and math.isfinite(interval[1][0]))])
+        birth = np.asarray([(interval[1][0]) for interval in persistence_dim if (isfinite(interval[1][1]) and isfinite(interval[1][0]))])
+        death = np.asarray([(interval[1][1]) for interval in persistence_dim if (isfinite(interval[1][1]) and isfinite(interval[1][0]))])
```

Comment:
If you start from a numpy array a, you can keep only the rows with a finite second column using `a[np.isfinite(a[:,1])]`.

Reply:
In the new version 7e8685458f684f347b2aa6a77a288c249c722f25 I tried to use numpy array as much as possible. It makes a 0.3 out of 4 seconds improvement for plot_persistence_density over 740 intervals pairs.